swap_meet/vendor.py

```diff
@@ -1,2 +1,99 @@
+import pytest
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    #return a list of items that the vendor has in a given category  
+    def get_by_category(self, category):
+        category_list = list(filter(lambda item: item.category == category, self.inventory))
+      
+        # category_list = []
+        # for item in self.inventory:
+        #     if item.category == category:
+        #         category_list.append(item)
+
+        return category_list
+
+    """
+    Swap_items vendor method checks if the items are in each vendor's inventory,
+    then removes and adds items from both vendor's inventories.
+    """
+    def swap_items(self, other_vendor, my_item, their_item):
+        
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        else:
```

Comment:
This else is not necessary! Since the if block has a return statement, if we move past it it's assumed we're in the else.